packages/react/src/primitives/types/phoneNumberField.ts

```diff
@@ -0,0 +1,6 @@
+import { TextFieldProps } from './textField';
+
+export interface PhoneNumberFieldProps extends TextFieldProps {
+  defaultCountryCode: string;
+  type?: 'tel';
```

Comment:
This is really where I wish we could use `Omit`, because we don't actually allow the customer to set the type on this primitive. This is a good solution for now.

Reply:
@reesscot I agree, I originally wanted to prevent `type` from being passed as a prop to the primitive entirely. I kind of got backed into a corner though since we're ultimately extending `BaseComponentProps` which has a catch all `[key: string]: any;` prop. So this was what I came up with to prevent the wrong type from being passed.

Another idea is I could use `Omit` like you suggested, pull out the `type` prop when destructuring arguments to `PhoneNumberField`, then force `type="tel"` so it doesn't get overwritten by a custom `type` attribute. Neither are entirely elegant though, so I'll leave this unless you have a preference.

Reply:
I wanted to do some cheekiness like 
```typescript
interface PhoneNumberFieldProps extends
  Omit<TextFieldProps, 'onChange'>,
  Pick<SelectFieldProps, 'onChange'> {
...
```
but then discovered that `Pick` and `Omit` utility types don't actually do deep inheritance. I'm sure you all have come across that and is why we aren't using those types often.

Reply:
They actually should work, but there's a problem with our BaseComponent type which makes Omit and Pick not work correclty. TLDR is that Omit does a union of all the keys that forces everything to `[key: string]: any`.

Reply:
Oh ok maybe that was the issue I was running into. It would be nice to be able to use these utility types to build out different primitives based on existing typings.